tests/main/snap-service-watchdog/task.yaml

```diff
@@ -0,0 +1,26 @@
+summary: Check that snaps can use software-watchdog provided by systemd
+
+execute: |
+    echo "When the service snap  is installed"
+    . $TESTSLIB/snaps.sh
+    install_local test-snapd-service-watchdog
+
+    echo "We can see all services running"
+    for service in direct-watchdog-ok direct-watchdog-bad; do
+        ! systemctl show -p SubState snap.test-snapd-service-watchdog.$service | MATCH "SubState=dead"
+    done
+
+    echo "Services that are correctly poking the watchdog remain running"
+    for i in $(seq 20); do
+        for service in direct-watchdog-ok; do
+            systemctl show -p SubState snap.test-snapd-service-watchdog.$service | MATCH 'SubState=running'
+        done
+        sleep 1
```

Comment:
That's 20 seconds dead on a single test that isn't doing much. We don't need to repeat the test several times I think, and we don't need to sleep also. We just need to make sure that one or two pings indeed worked, and that one restart due lack of ping indeed worked. We can do that under one second if we use a short ping time.